HW04.Rmd

```diff
@@ -1,126 +1,186 @@
+
+```{r month_chart}
+#Originally variable was read as an integer, so I changed its class to be read as a factor
+gun_deaths$month <- as.factor(gun_deaths$month)
+
+#Simple barplot with the number of deaths per month
+ggplot(gun_deaths,mapping = aes(x = month)) +
+  geom_bar() +   
+  scale_x_discrete(labels=c("Jan","Feb","Mar","Apr","May","Jun","Jul",
+                            "Aug","Sep","Oct","Nov","Dec"))+
+  theme_bw()
+```
+
+## Generating a bar chart that identifies the number of gun deaths associated with each type of intent cause of death. The bars should be sorted from highest to lowest values.
+
+```{r intent}
+#Barplot with number of deaths by intent (sorted from highest to lowest values)
+#Sorted using the function fct_infreq from the forcats package
+ggplot(gun_deaths,mapping = aes(x = forcats::fct_infreq(intent))) +
+  geom_bar(col="gray30",fill="gray70")+
+  theme_bw()+
+  theme(axis.title = element_text(color="black", size=12))
+```
+
+## Generating a boxplot visualizing the age of gun death victims, by sex. Print the average age of female gun death victims.
+
+```{r age_&_sex}
+#I want the age to be read as a numeric variable.
+gun_deaths$age <- as.numeric(gun_deaths$age)
+
+#Boxplot showing the age of gun death victims by sex:
+ggplot(gun_deaths,mapping = aes(x = sex, y= age)) +
+  geom_boxplot(fill="gray90",width=0.7)+
+  theme_bw()+
+  theme(axis.title = element_text(color="black", size=12))
+
+#This is the mean of the age of female gun death victims:
+F_data <- gun_deaths %>% filter(sex=="F")
+mean(F_data$age)
+```
+
+## How many white males with at least a high school education were killed by guns in 2012?
+```{r education}
+#The answer is:
+kable(gun_deaths %>% filter(sex=="M") %>% filter(race=="White") %>% filter(year=="2012") %>% filter(education !="Less than HS")%>% count(), format="markdown")
+```
+
+## Which season of the year has the most gun deaths?
+
+Assume that:
+
+* Winter = January-March
+* Spring = April-June
+* Summer = July-September
+* Fall = October-December
+
+```{r season}
+#First, I created an additional column (variable) that specifies the season:
+gun_deaths$season[gun_deaths$month %in% c("1","2","3")] <- "winter"
+gun_deaths$season[gun_deaths$month %in% c("4","5","6")] <- "spring"
+gun_deaths$season[gun_deaths$month %in% c("7","8","9")] <- "summer"
+gun_deaths$season[gun_deaths$month %in% c("10","11","12")] <- "fall"
+
+#Then, I created a summary table counting the number of deaths per season. The season with the highest numer of deaths due tu gun violence is summer:
+kable(gun_deaths %>% count(season, sort=TRUE),format="markdown",
+      col.names=c("Season","Numer of gun deaths"))
+```
+
+# These are more open ended questions from here on out, you could look at making multiple summaries or types of plots. 
+
+### Are whites who are killed by guns more likely to die because of suicide or homicide? How does this compare to blacks and hispanics?
```

Comment:
I got slightly different percentages than you, but I filtered out the accidental, undetermined and not available causes of death, so it's probably just that.